cartography/models/aws/ec2/networkinterface_instance.py

```diff
@@ -0,0 +1,109 @@
+from dataclasses import dataclass
+
+from cartography.models.core.common import PropertyRef
+from cartography.models.core.nodes import CartographyNodeProperties
+from cartography.models.core.nodes import CartographyNodeSchema
+from cartography.models.core.relationships import CartographyRelProperties
+from cartography.models.core.relationships import CartographyRelSchema
+from cartography.models.core.relationships import LinkDirection
+from cartography.models.core.relationships import make_target_node_matcher
+from cartography.models.core.relationships import OtherRelationships
+from cartography.models.core.relationships import TargetNodeMatcher
+
+
+@dataclass(frozen=True)
+class EC2NetworkInterfaceInstanceNodeProperties(CartographyNodeProperties):
+    """
+    Selection of properties of a network interface as known by an EC2 instance
+    """
+    # arn: PropertyRef = PropertyRef('Arn', extra_index=True) TODO use arn; issue #1024
+    id: PropertyRef = PropertyRef('NetworkInterfaceId')
+    status: PropertyRef = PropertyRef('Status')
+    mac_address: PropertyRef = PropertyRef('MacAddress', extra_index=True)
+    description: PropertyRef = PropertyRef('Description')
+    private_dns_name: PropertyRef = PropertyRef('PrivateDnsName', extra_index=True)
+    private_ip_address: PropertyRef = PropertyRef('PrivateIpAddress', extra_index=True)
```

Comment:
@serge-wq - [comment 1 of 2] here's an example of what we were talking about earlier: a network interface as known by an EC2 instance has a private_ip_address...

Reply:
actually never mind, wait let me find a better example